### PR TITLE
Safe Secrets KPI: Convert Storage Account Connection Strings to MSI

### PR DESCRIPTION
**Pull request by gauravIoTEdge**
***Please replace this line with your PR description and read PR checklist below***

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: test/modules/TestResultCoordinator/TestReportUtil.cs [hunk 1]
```diff
@@ -128,10 +129,15 @@ internal static List<ITestReportMetadata> ParseReportMetadataJson(string reportM
             return reportMetadataList;
         }
 
-        internal static async Task<Uri> GetOrCreateBlobContainerSasUriForLogAsync(string storageAccountConnectionString)
+        internal static async Task<Uri> GetOrCreateBlobContainerSasUriForLogAsync(string storageAccountName)
         {
-            string containerName = GetAzureBlobContainerNameForLog();
-            var containerClient = new BlobContainerClient(storageAccountConnectionString, containerName);
+            // TODO: Replace <storage-account-name> with your actual storage account name
```
Comment (damonbarry) [inline test/modules/TestResultCoordinator/TestReportUtil.cs:134]: (Apologies if this review is premature)
Remove TODO comment

File: builds/e2e/connectivity.yaml [hunk 2]
```diff
@@ -169,6 +168,46 @@ jobs:
...
+      - task: AzureCLI@2
+        displayName: 'Get SAS URI for Blob Storage Account'
+        inputs:
+          azureSubscription: $(azure.subscription)
+          scriptType: 'bash'
+          scriptLocation: 'inlineScript'
+          inlineScript: |
+            # Hardcoded resource group and storage account name
+            resource_group="iotedge-deploy"
+            storage_account="iotedgeconnectivitytest"
```
Comment (yophilav) [inline builds/e2e/connectivity.yaml:180]: Please make these task variables.

File: builds/e2e/connectivity.yaml [hunk 3]
```diff
@@ -169,6 +168,46 @@ jobs:
...
+          
+            # Set container name with current date
+            container_name="logs$(date +%Y%m%d)"
+            
+            # Set expiry time for the SAS token
+            expiry_time=$(date --utc --date "+2 days" +"%Y-%m-%dT%H:%MZ")
```
Comment (yophilav) [inline builds/e2e/connectivity.yaml:186]: Please put `"`around the `$()`

File: builds/e2e/connectivity.yaml [hunk 4]
```diff
@@ -169,6 +168,46 @@ jobs:
...
+            
+            # Generate the SAS token for the container
+            sasToken=$(az storage container generate-sas --account-name "$storage_account" --name "$container_name" --https-only --expiry "$expiry_time" --as-user --auth-mode login --permissions drwl --output tsv)
```
Comment (yophilav) [inline builds/e2e/connectivity.yaml:189]: Please put `"`around the `$()`

File: scripts/linux/trcE2ETest.sh [hunk 5]
```diff
@@ -917,7 +919,7 @@ function configure_connectivity_settings() {
 
     TEST_INFO="$TEST_INFO,TestDuration=${TEST_DURATION}"
 
-    log_upload_enabled=false
+    log_upload_enabled=true
```
Comment (yophilav) [inline scripts/linux/trcE2ETest.sh:922]: We don't need to upload log for most cases. Only when you're testing the upload (or if the test result fails.)

File: builds/e2e/connectivity.yaml (hunk 2 above)
Comment (damonbarry) [reply builds/e2e/connectivity.yaml:180]: @yophilav task variables, as in, `##vso[task.setvariable ...]`? These are only needed within the bash script, so bash vars are fine.

**Comment by gauravIoTEdge** [comment]
superseded by other pull requests

**Pull request closed without merging**